2024-10-06 23:59:54,883 INFO     "172.19.0.1" - - [06/Oct/2024:23:59:54 +0000] "GET / HTTP/1.1" 200 1782 "-" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-06 23:59:55,137 INFO     "172.19.0.1" - - [06/Oct/2024:23:59:54 +0000] "GET /static/app/datatables.css HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-06 23:59:55,142 INFO     "172.19.0.1" - - [06/Oct/2024:23:59:54 +0000] "GET /static/app/bootstrap-5.3.3-dist/css/bootstrap.min.css HTTP/1.1" 200 232803 "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-06 23:59:55,147 INFO     "172.19.0.1" - - [06/Oct/2024:23:59:54 +0000] "GET /static/app/bridge.css HTTP/1.1" 200 372 "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-06 23:59:55,162 INFO     "172.19.0.1" - - [06/Oct/2024:23:59:54 +0000] "GET /static/app/bootstrap-5.3.3-dist/js/bootstrap.bundle.js HTTP/1.1" 200 207819 "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-06 23:59:55,163 INFO     "172.19.0.1" - - [06/Oct/2024:23:59:54 +0000] "GET /static/django_eventstream/eventsource.min.js HTTP/1.1" 200 10353 "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-06 23:59:55,164 INFO     "172.19.0.1" - - [06/Oct/2024:23:59:54 +0000] "GET /static/django_eventstream/reconnecting-eventsource.js HTTP/1.1" 200 3541 "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-06 23:59:55,175 INFO     "172.19.0.1" - - [06/Oct/2024:23:59:54 +0000] "GET /static/app/datatables.js HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-06 23:59:55,176 INFO     "172.19.0.1" - - [06/Oct/2024:23:59:54 +0000] "GET /static/app/sse.js HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-06 23:59:55,180 INFO     "172.19.0.1" - - [06/Oct/2024:23:59:54 +0000] "GET /static/app/htmx.js HTTP/1.1" 200 160809 "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-06 23:59:59,719 INFO     "172.19.0.1" - - [06/Oct/2024:23:59:59 +0000] "GET /accounts/login/ HTTP/1.1" 200 2648 "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 00:00:04,741 INFO     "172.19.0.1" - - [07/Oct/2024:00:00:04 +0000] "POST /accounts/login/ HTTP/1.1" 302 - "https://teensy-info.tail571dc2.ts.net/accounts/login/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 00:00:04,911 INFO     "172.19.0.1" - - [07/Oct/2024:00:00:04 +0000] "GET / HTTP/1.1" 200 3147 "https://teensy-info.tail571dc2.ts.net/accounts/login/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 00:00:08,144 INFO     "172.19.0.1" - - [07/Oct/2024:00:00:07 +0000] "GET /player/1/ HTTP/1.1" 200 6602 "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 00:00:08,432 INFO     "172.19.0.1" - - [07/Oct/2024:00:00:07 +0000] "GET /events/player/players:1_1 HTTP/1.1" 301 - "https://teensy-info.tail571dc2.ts.net/player/1/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 00:00:10,189 INFO     "172.19.0.1" - - [07/Oct/2024:00:00:09 +0000] "GET /table/15/ HTTP/1.1" 302 - "https://teensy-info.tail571dc2.ts.net/player/1/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 00:00:12,866 INFO     {'contract_points': 0, 'overtrick_points': 0, 'slam_bonus': 0, 'doubled_bonus': 0, 'penalty_points': 100, 'game_bonus': 0, 'east_west_tricks': 7, 'north_south_tricks': 6}
2024-10-07 00:00:12,866 INFO     score_description='declarers got -100 or I suppose you could say defenders got 100'
2024-10-07 00:00:12,984 INFO     "172.19.0.1" - - [07/Oct/2024:00:00:12 +0000] "GET /events/player/players:1_1 HTTP/1.1" 301 - "https://teensy-info.tail571dc2.ts.net/player/1/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 00:00:12,989 INFO     "172.19.0.1" - - [07/Oct/2024:00:00:12 +0000] "GET /table/15/archive/ HTTP/1.1" 200 2605 "https://teensy-info.tail571dc2.ts.net/player/1/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 00:01:12,985 INFO     Timing out client: IPv4Address(type='TCP', host='172.19.0.1', port=45084)
2024-10-07 00:01:12,990 INFO     Timing out client: IPv4Address(type='TCP', host='172.19.0.1', port=45072)
2024-10-07 03:31:59,069 INFO     "172.19.0.1" - - [07/Oct/2024:03:31:58 +0000] "GET / HTTP/1.1" 200 3147 "-" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:31:59,248 INFO     "172.19.0.1" - - [07/Oct/2024:03:31:58 +0000] "GET /static/app/datatables.css HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:31:59,257 INFO     "172.19.0.1" - - [07/Oct/2024:03:31:58 +0000] "GET /static/app/bootstrap-5.3.3-dist/css/bootstrap.min.css HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:31:59,258 INFO     "172.19.0.1" - - [07/Oct/2024:03:31:58 +0000] "GET /static/app/bridge.css HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:31:59,276 INFO     "172.19.0.1" - - [07/Oct/2024:03:31:59 +0000] "GET /static/django_eventstream/eventsource.min.js HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:31:59,279 INFO     "172.19.0.1" - - [07/Oct/2024:03:31:59 +0000] "GET /static/app/bootstrap-5.3.3-dist/js/bootstrap.bundle.js HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:31:59,304 INFO     "172.19.0.1" - - [07/Oct/2024:03:31:59 +0000] "GET /static/django_eventstream/reconnecting-eventsource.js HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:31:59,306 INFO     "172.19.0.1" - - [07/Oct/2024:03:31:59 +0000] "GET /static/app/datatables.js HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:31:59,307 INFO     "172.19.0.1" - - [07/Oct/2024:03:31:59 +0000] "GET /static/app/htmx.js HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:31:59,309 INFO     "172.19.0.1" - - [07/Oct/2024:03:31:59 +0000] "GET /static/app/sse.js HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:32:04,095 INFO     "172.19.0.1" - - [07/Oct/2024:03:32:03 +0000] "GET /player/1/ HTTP/1.1" 200 6602 "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:32:04,350 INFO     "172.19.0.1" - - [07/Oct/2024:03:32:04 +0000] "GET /events/player/players:1_1 HTTP/1.1" 301 - "https://teensy-info.tail571dc2.ts.net/player/1/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:32:10,298 INFO     "172.19.0.1" - - [07/Oct/2024:03:32:10 +0000] "GET /player/1/ HTTP/1.1" 200 6484 "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:32:10,442 ERROR    Future exception was never retrieved
future: <Future finished exception=CancelledError() created at /usr/local/lib/python3.12/asyncio/base_events.py:449>
source_traceback: Object created at (most recent call last):
  File "/bridge/.venv/bin/daphne", line 8, in <module>
    sys.exit(CommandLineInterface.entrypoint())
  File "/bridge/.venv/lib/python3.12/site-packages/daphne/cli.py", line 171, in entrypoint
    cls().run(sys.argv[1:])
  File "/bridge/.venv/lib/python3.12/site-packages/daphne/cli.py", line 285, in run
    self.server.run()
  File "/bridge/.venv/lib/python3.12/site-packages/daphne/server.py", line 150, in run
    reactor.run(installSignalHandlers=self.signal_handlers)
  File "/bridge/.venv/lib/python3.12/site-packages/twisted/internet/asyncioreactor.py", line 253, in run
    self._asyncioEventloop.run_forever()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 641, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1978, in _run_once
    handle._run()
  File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "/bridge/.venv/lib/python3.12/site-packages/django/core/handlers/asgi.py", line 185, in process_request
    response = await self.run_get_response(request)
  File "/bridge/.venv/lib/python3.12/site-packages/django/core/handlers/asgi.py", line 244, in run_get_response
    response = await self.get_response_async(request)
  File "/bridge/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/django/asgi.py", line 129, in sentry_patched_get_response_async
    return await old_get_response_async(self, request)
  File "/bridge/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 162, in get_response_async
    response = await self._middleware_chain(request)
  File "/bridge/.venv/lib/python3.12/site-packages/asgiref/sync.py", line 455, in __call__
    exec_coro = loop.run_in_executor(
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 862, in run_in_executor
    return futures.wrap_future(
  File "/usr/local/lib/python3.12/asyncio/futures.py", line 421, in wrap_future
    new_future = loop.create_future()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 449, in create_future
    return futures.Future(loop=self)
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/asgiref/sync.py", line 522, in thread_handler
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/django/middleware.py", line 173, in __call__
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/debug_toolbar/middleware.py", line 72, in __call__
    return self.get_response(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/django/middleware.py", line 173, in __call__
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 129, in __call__
    response = response or self.get_response(request)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/django/middleware.py", line 173, in __call__
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/whitenoise/middleware.py", line 124, in __call__
    return self.get_response(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/asgiref/sync.py", line 254, in __call__
    return call_result.result()
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/_base.py", line 449, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
  File "/bridge/.venv/lib/python3.12/site-packages/asgiref/sync.py", line 331, in main_wrap
    result = await self.awaitable(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 42, in inner
    response = await get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/django/asgi.py", line 232, in __acall__
    return await f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 145, in __acall__
    response = response or await self.get_response(request)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 42, in inner
    response = await get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/django/asgi.py", line 232, in __acall__
    return await f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 145, in __acall__
    response = response or await self.get_response(request)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 42, in inner
    response = await get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/django/asgi.py", line 232, in __acall__
    return await f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 141, in __acall__
    response = await sync_to_async(
               ^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/asgiref/sync.py", line 468, in __call__
    ret = await asyncio.shield(exec_coro)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
asyncio.exceptions.CancelledError
2024-10-07 03:32:10,651 INFO     "172.19.0.1" - - [07/Oct/2024:03:32:10 +0000] "GET /events/player/players:1_1 HTTP/1.1" 301 - "https://teensy-info.tail571dc2.ts.net/player/1/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:32:11,653 INFO     "172.19.0.1" - - [07/Oct/2024:03:32:11 +0000] "GET /players/?lookin_for_love=True HTTP/1.1" 200 6663 "https://teensy-info.tail571dc2.ts.net/player/1/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:32:13,324 INFO     "172.19.0.1" - - [07/Oct/2024:03:32:13 +0000] "GET /player/44/ HTTP/1.1" 200 6873 "https://teensy-info.tail571dc2.ts.net/players/?lookin_for_love=True" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:32:13,568 INFO     "172.19.0.1" - - [07/Oct/2024:03:32:13 +0000] "GET /events/player/players:1_44 HTTP/1.1" 301 - "https://teensy-info.tail571dc2.ts.net/player/44/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:32:14,549 INFO     "172.19.0.1" - - [07/Oct/2024:03:32:14 +0000] "POST /player/44/ HTTP/1.1" 200 - "https://teensy-info.tail571dc2.ts.net/player/44/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:32:14,652 INFO     "172.19.0.1" - - [07/Oct/2024:03:32:14 +0000] "GET /player/44/partnership/ HTTP/1.1" 200 810 "https://teensy-info.tail571dc2.ts.net/player/44/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:32:15,759 INFO     "172.19.0.1" - - [07/Oct/2024:03:32:15 +0000] "GET /players/?seated=False&lookin_for_love=False&exclude_me=True HTTP/1.1" 200 5931 "https://teensy-info.tail571dc2.ts.net/player/44/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:32:16,975 ERROR    Internal Server Error: /table/new/32/1/
Traceback (most recent call last):
  File "/bridge/.venv/lib/python3.12/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "app_handrecord_board_id_key"
DETAIL:  Key (board_id)=(1) already exists.


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/bridge/.venv/lib/python3.12/site-packages/asgiref/sync.py", line 518, in thread_handler
    raise exc_info[1]
  File "/bridge/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 42, in inner
    response = await get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/asgiref/sync.py", line 518, in thread_handler
    raise exc_info[1]
  File "/bridge/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 253, in _get_response_async
    response = await wrapped_callback(
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/asgiref/sync.py", line 468, in __call__
    ret = await asyncio.shield(exec_coro)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/asgiref/current_thread_executor.py", line 40, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/asgiref/sync.py", line 522, in thread_handler
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/views/decorators/http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/contrib/auth/decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/project/app/views/misc.py", line 42, in non_players_piss_off
    return view_function(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/project/app/views/table/details.py", line 495, in new_table_for_two_partnerships
    t = app.models.Table.objects.create_with_two_partnerships(p1, p2)  # type: ignore
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/project/app/models/table.py", line 70, in create_with_two_partnerships
    Hand.objects.create(board=b, table=t)
  File "/bridge/.venv/lib/python3.12/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/db/models/query.py", line 679, in create
    obj.save(force_insert=True, using=self.db)
  File "/bridge/.venv/lib/python3.12/site-packages/django/db/models/base.py", line 891, in save
    self.save_base(
  File "/bridge/.venv/lib/python3.12/site-packages/django/db/models/base.py", line 997, in save_base
    updated = self._save_table(
              ^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/db/models/base.py", line 1160, in _save_table
    results = self._do_insert(
              ^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/db/models/base.py", line 1201, in _do_insert
    return manager._insert(
           ^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/db/models/query.py", line 1847, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/db/models/sql/compiler.py", line 1836, in execute_sql
    cursor.execute(sql, params)
  File "/bridge/.venv/lib/python3.12/site-packages/sentry_sdk/utils.py", line 1761, in runner
    return sentry_patched_function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/django/__init__.py", line 651, in execute
    result = real_execute(self, sql, params)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/db/backends/utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/db/backends/utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/db/backends/utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/bridge/.venv/lib/python3.12/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.IntegrityError: duplicate key value violates unique constraint "app_handrecord_board_id_key"
DETAIL:  Key (board_id)=(1) already exists.

2024-10-07 03:32:16,995 INFO     "172.19.0.1" - - [07/Oct/2024:03:32:16 +0000] "POST /table/new/32/1/ HTTP/1.1" 500 145 "https://teensy-info.tail571dc2.ts.net/players/?seated=False&lookin_for_love=False&exclude_me=True" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 03:33:16,997 INFO     Timing out client: IPv4Address(type='TCP', host='172.19.0.1', port=40982)
2024-10-07 12:48:40,480 WARNING  Executing <Handle callWithLogger(<<class 'twis...ory'> on 9000>, <bound method...f9211be31dc0>>, <<class 'twis...ory'> on 9000>, True) at /bridge/.venv/lib/python3.12/site-packages/twisted/python/log.py:83> took 0.128 seconds
2024-10-07 12:48:40,793 WARNING  Executing <Handle callWithLogger(<_GenericHTTP...l #39 on 9000>, <bound method...f9211be31dc0>>, <_GenericHTTP...l #39 on 9000>, True) at /bridge/.venv/lib/python3.12/site-packages/twisted/python/log.py:83 created at /usr/local/lib/python3.12/asyncio/selector_events.py:280> took 0.273 seconds
2024-10-07 12:48:43,018 WARNING  Forbidden: /table/new/32/1/
2024-10-07 12:48:43,067 INFO     "172.19.0.1" - - [07/Oct/2024:12:48:42 +0000] "POST /table/new/32/1/ HTTP/1.1" 403 - "https://teensy-info.tail571dc2.ts.net/players/?seated=False&lookin_for_love=False&exclude_me=True" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 12:49:07,146 INFO     "172.19.0.1" - - [07/Oct/2024:12:49:06 +0000] "GET / HTTP/1.1" 200 3147 "-" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 12:49:07,472 INFO     "172.19.0.1" - - [07/Oct/2024:12:49:06 +0000] "GET /static/app/bootstrap-5.3.3-dist/css/bootstrap.min.css HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 12:49:07,518 INFO     "172.19.0.1" - - [07/Oct/2024:12:49:07 +0000] "GET /static/app/datatables.css HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 12:49:07,519 INFO     "172.19.0.1" - - [07/Oct/2024:12:49:07 +0000] "GET /static/app/bridge.css HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 12:49:07,533 INFO     "172.19.0.1" - - [07/Oct/2024:12:49:07 +0000] "GET /static/app/htmx.js HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 12:49:07,535 INFO     "172.19.0.1" - - [07/Oct/2024:12:49:07 +0000] "GET /static/app/bootstrap-5.3.3-dist/js/bootstrap.bundle.js HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 12:49:07,542 INFO     "172.19.0.1" - - [07/Oct/2024:12:49:07 +0000] "GET /static/app/datatables.js HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 12:49:07,543 INFO     "172.19.0.1" - - [07/Oct/2024:12:49:07 +0000] "GET /static/app/sse.js HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 12:49:07,546 INFO     "172.19.0.1" - - [07/Oct/2024:12:49:07 +0000] "GET /static/django_eventstream/eventsource.min.js HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 12:49:07,564 INFO     "172.19.0.1" - - [07/Oct/2024:12:49:07 +0000] "GET /static/django_eventstream/reconnecting-eventsource.js HTTP/1.1" 304 - "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 12:49:09,434 INFO     "172.19.0.1" - - [07/Oct/2024:12:49:08 +0000] "GET /player/1/ HTTP/1.1" 200 6602 "https://teensy-info.tail571dc2.ts.net/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 12:49:09,774 INFO     "172.19.0.1" - - [07/Oct/2024:12:49:09 +0000] "GET /events/player/players:1_1 HTTP/1.1" 301 - "https://teensy-info.tail571dc2.ts.net/player/1/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
2024-10-07 12:49:13,366 ERROR    Internal Server Error: /table/16/
Traceback (most recent call last):
  File "/bridge/.venv/lib/python3.12/site-packages/asgiref/sync.py", line 518, in thread_handler
    raise exc_info[1]
  File "/bridge/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 42, in inner
    response = await get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/asgiref/sync.py", line 518, in thread_handler
    raise exc_info[1]
  File "/bridge/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 253, in _get_response_async
    response = await wrapped_callback(
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/asgiref/sync.py", line 468, in __call__
    ret = await asyncio.shield(exec_coro)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/asgiref/current_thread_executor.py", line 40, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/asgiref/sync.py", line 522, in thread_handler
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/utils/decorators.py", line 188, in _view_wrapper
    result = _process_exception(request, e)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/utils/decorators.py", line 186, in _view_wrapper
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/contrib/auth/decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/project/app/views/misc.py", line 42, in non_players_piss_off
    return view_function(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/project/app/views/table/details.py", line 465, in table_detail_view
    _four_hands_context_for_table(request, table)
  File "/bridge/project/app/views/table/details.py", line 263, in _four_hands_context_for_table
    skel = table.display_skeleton(as_dealt=as_dealt)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/bridge/project/app/models/table.py", line 248, in display_skeleton
    xscript = self.current_hand.xscript
              ^^^^^^^^^^^^^^^^^
  File "/bridge/.venv/lib/python3.12/site-packages/django/utils/functional.py", line 47, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
                                         ^^^^^^^^^^^^^^^^^^^
  File "/bridge/project/app/models/table.py", line 207, in current_hand
    assert rv is not None
           ^^^^^^^^^^^^^^
AssertionError
2024-10-07 12:49:13,439 INFO     "172.19.0.1" - - [07/Oct/2024:12:49:12 +0000] "GET /table/16/ HTTP/1.1" 500 145 "https://teensy-info.tail571dc2.ts.net/player/1/" "Mozilla/5.0 (Android 14; Mobile; rv:131.0) Gecko/131.0 Firefox/131.0"
